packages/flame/test/components/priority_test.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:test/test.dart';
+
+class PriorityComponent extends BaseComponent {
+  PriorityComponent(int priority) : super(priority: priority);
+}
+
+bool isSorted<T>(Iterable<T> list, [int Function(T, T)? compare]) {
```

Comment:
do we need a generic isSorted? we could just do
```
p = components.map(e => e.priority)
expect(p, orderedEquals(p.sorted()))
```

Reply:
(assuming we only ever care about List<Component>, otherwise the func is much more generic

Reply:
Much better idea, fixed that now